derive/src/decode.rs

```diff
@@ -300,3 +285,48 @@ fn create_instance(
 		},
 	}
 }
+
+pub fn quote_decode_with_mem_tracking_checks(data: &Data, crate_path: &syn::Path) -> TokenStream {
+	let fields: Box<dyn Iterator<Item = &Field>> = match data {
+		Data::Struct(data) => Box::new(data.fields.iter()),
+		Data::Enum(ref data) => {
+			let variants = match utils::try_get_variants(data) {
+				Ok(variants) => variants,
+				Err(e) => return e.to_compile_error(),
+			};
+
+			let mut fields: Box<dyn Iterator<Item = &Field>> = Box::new(iter::empty());
+			for variant in variants {
+				fields = Box::new(fields.chain(variant.fields.iter()));
```

Comment:
Why do you use an iter here instead of a vector?

Reply:
It should be more efficient. If we want to build a vector we have to iterate through the fields and `collect()`. This way we just get the iterator and use it below.

Reply:
Huh, derive macros dont have to be optimized, but okay.